src/react-apps/applications/shared/src/version-control/versionControlHeader.tsx

```diff
@@ -223,7 +223,17 @@ class VersionControlHeader extends React.Component<IVersionControlHeaderProps, I
     });
   }
 
-  public shareChanges = (currentTarget: any) => {
+  public shareChanges = (currentTarget: any, showNothingToPush: boolean) => {
+    if (showNothingToPush) {
+      this.setState({
+        anchorEl: currentTarget,
+        modalState: {
+          shouldShowDoneIcon: true,
+          header: getLanguageFromKey('sync_header.nothing_to_push', this.props.language),
+        },
+      });
+      return;
```

Comment:
Instead of having `return` on its own line, like an orphan at Christmas.
Just do:
```
return this.setState({
  anchorEl: currentTarget,
  modalState: {
    shouldShowDoneIcon: true,
    header: getLanguageFromKey('sync_header.nothing_to_push', this.props.language),
  },
});
```